panoply/datasource.py

```diff
@@ -18,13 +24,58 @@ def log(self, *msgs):
         if "logger" in self.options:
             self.options["logger"](msgs)
         else:
-            print msgs
+            print(msgs)
 
-    def progress(self, loaded, total, msg):
+    def progress(self, loaded, total, msg=''):
         self.fire("progress", {
             "loaded": loaded,
             "total": total,
             "msg": msg
         })
 
+# a decorator used to invalidate the access_token for oauth based data sources
+# this should be used on every method in the data source that fetches data
+# from the server (controlled by this oauth), and that needs to be invalidated
+def invalidate_token(refresh_url, callback=None,
```

Comment:
I don't like the side effect of this function. Actually, all of it is just a side-effect I never can tell what will be the state of the source after this function was invoked.
We should consider exposing a function that will get the url and the refresh_token and will return a new refreshed token.